plugins/additional/additional.cpp

```diff
@@ -1320,12 +1320,12 @@ class SearchCtrlComponent : public ComponentBase
 
 		if ( !obj->IsNull( _("search_button") ) )
 		{
-			sc->ShowSearchButton( obj->GetPropertyAsInteger( _("search_button") ) );
+			sc->ShowSearchButton( !!obj->GetPropertyAsInteger( _("search_button") ) );
```

Comment:
I think `!= 0` would be clearer.

Reply:
You can debate if the double bang is clearer, but it is shorter and more flexible to normalize X to a bool. It can also be used in cases where X can't be compared to an int, e.g. the safe bool idiom or a ``wxStreamBase`` that doesn't convert to a bool at all and only has ``operator!``.

Reply:
In this case it actually *is* an integer though.

Reply:
Still it is shorter and has its semantic meaning of "normalize-to-bool-operator". I am used to use it in my codebase, so what shall i do?

Reply:
If it isn't okay to implicitly cast an int to bool, I don't think that passing an int to `!` should be allowed either.

> safe bool idiom

If I understand it correctly this is obsolete now that C++11 supports explicit conversion operators.

Reply:
This doesn't stop [MSVC](https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-3-c4800) to complain about

Reply:
I see you don't like my double bang operator, i could go the /wd4800 route if you prefer

Reply:
> This doesn't stop MSVC to complain about

What do you mean by "this"?

> I see you don't like my double bang operator, i could go the /wd4800 route if you prefer

I still think `!= 0` would be a good idea, it will also silence MSVC's warning. I will push something tomorrow.

Reply:
> What do you mean by "this"?

Explicit conversion